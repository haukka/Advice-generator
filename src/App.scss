.generator-app {
  height: 100%;
  background-position: center;
  background-size: cover;
  background: linear-gradient( rgba(0, 0, 0, 0.3), rgba(0, 0, 0, 0.3) ), url('../public/petit-prince.jpg');
  display: flex;
  justify-content: center;
  align-items: center;
}

.card {
  min-height: 20%;
  flex-basis: 40%;
  display: flex;
  background: whitesmoke;
  flex-direction: column;
  align-items: center;
  border-radius: 25px;
  padding: 2%;
  box-shadow: 10px 10px;
}

.advice {
  display: flex;
  align-items: center;
  font-family: 'Spartan', 'sans-serif';
}

.button {
  position: relative;
  outline: none;
  text-decoration: none;
  border-radius: 50px;
  display: flex;
  justify-content: center;
  align-items: center;
  cursor: pointer;
  text-transform: uppercase;
  max-height: 30px;
  flex-basis: 210px;
  opacity: 1;
  background-color: #ffffff;
  border: 1px solid rgba(22, 76, 167, 0.6);
}
  
.button span {
  color: #164ca7;
  font-size: 12px;
  font-weight: 500;
  letter-spacing: 0.7px;
}
  
.button:hover {
  animation: rotate 0.7s ease-in-out both;
}
  
.button:hover span {
  animation: storm 0.7s ease-in-out both;
  animation-delay: 0.06s;
}

button:disabled,
button[disabled]{
  border: 1px solid #999999;
  background-color: #cccccc;
  color: #666666;
}
  
@keyframes rotate {
  0% {
    transform: rotate(0deg) translate3d(0, 0, 0);
  }
  25% {
    transform: rotate(3deg) translate3d(0, 0, 0);
  }
  50% {
    transform: rotate(-3deg) translate3d(0, 0, 0);
  }
  75% {
    transform: rotate(1deg) translate3d(0, 0, 0);
  }
  100% {
    transform: rotate(0deg) translate3d(0, 0, 0);
  }
}
  
@keyframes storm {
  0% {
    transform: translate3d(0, 0, 0) translateZ(0);
  }
  25% {
    transform: translate3d(4px, 0, 0) translateZ(0);
  }
  50% {
    transform: translate3d(-3px, 0, 0) translateZ(0);
  }
  75% {
    transform: translate3d(2px, 0, 0) translateZ(0);
  }
  100% {
    transform: translate3d(0, 0, 0) translateZ(0);
  }
}  

@media only screen and (max-width: 600px) {
  .card {
      width: 80%;
      height: 30%;
  }
}