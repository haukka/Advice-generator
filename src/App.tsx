import React, { useEffect, useState } from 'react';
import axios from 'axios';
import './App.scss';
import AnimatedLetters from './AnimatedLetters';

const App = () => {
  const [advice, setAdvice] = useState('');
  const [letterClass, setLetterClass] = useState('text-animate');
  const [disabled, setDisabled] = useState(true);

  const fetchAdvice = async () => {
    try {
      setDisabled(true);
      const result = await axios.get('https://api.adviceslip.com/advice');
      const adviceResult = result?.data?.slip?.advice;
      setDisabled(false);
      setAdvice(adviceResult);
    } catch (error: any) {
      console.log(error);
    }
  }

  useEffect((): any => {
    const timer = setTimeout(() => {
      setLetterClass('text-animate-hover');
    }, 3000);
    return () => clearTimeout(timer);
  }, []);

  useEffect(() => {
    fetchAdvice();
  }, []);

  return (
    <div className='generator-app'>
      <div className='card'>
        <h1 className='advice'>
          {
            advice.length > 0 ? <span data-testid="animate-letter"><AnimatedLetters
              letterClass={letterClass}
              strArray={advice.split('')}
              idx={1}
            /></span> : null
          }
        </h1>
        <button data-testid="button-advice" disabled={disabled} className='button' onClick={fetchAdvice}>
          <span>Nouveau conseil</span>
        </button>
      </div>
    </div>
  );
}

export default App;
