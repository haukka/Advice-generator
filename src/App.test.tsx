import React from 'react';
import { act, cleanup, fireEvent, render, renderHook, screen, waitFor } from '@testing-library/react';
import App from './App';
import axiosMock from "axios";
import axios from 'axios';

afterEach(cleanup);

jest.mock('axios');
const mockedAxios = axios as jest.Mocked<typeof axios>;

describe("App", () => {

  it('test the presence of the button', () => {
    render(<App />);
    const button = screen.getByRole('button', { name: /Nouveau conseil/i });
    expect(button).toBeInTheDocument();
  });

  it('test axios', async () => {
    mockedAxios.get.mockResolvedValueOnce({ data: { slip: { advice: 'test' } } });
    render(<App />);
    const button = screen.getByRole('button', { name: /Nouveau conseil/i });

    // eslint-disable-next-line testing-library/no-unnecessary-act
    act(() => {
      fireEvent.click(button);
    });

    const resolvedSpan = await waitFor(() => screen.findByTestId('animate-letter'));
    expect(resolvedSpan).toBeInTheDocument();
    expect(axiosMock.get).toHaveBeenCalledTimes(1);
  });
});
