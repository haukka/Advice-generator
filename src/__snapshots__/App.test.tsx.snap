// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders learn react link 1`] = `Object {}`;
